### ISSUE TITLE
Rewrite interface "com.codename1.ui.list.ListModel" to use generics

### ISSUE THREAD

**Issue by ghost** [codenameone/CodenameOne#679]
Original [issue 679](https://code.google.com/p/codenameone/issues/detail?id=679) created by codenameone on 2013-04-27T14:10:46.000Z:

Hi,

I tried to implement the interface com.codename1.ui.list.ListModel a basic list class that can be reused in all of my lists.

I came across the limitation that you're using Object here all the way. What about using generics?

I rewrote the interface to work with generics and posted it in a gist together with a possible use-case:
https://gist.github.com/SimonSimCity/5473222

Here's a gist for the same use-case and what I had to do when the model isn't changed:
https://gist.github.com/SimonSimCity/5473110

Basically I have to do a lot more type-casts.

Bye
Simon
